web/src/components/apps/Dashboard.tsx

```diff
@@ -174,17 +263,17 @@ class Dashboard extends Component {
     });
   };
 
-  getAppDashboard = () => {
+  getAppDashboard = (): Promise<void> => {
     return new Promise((resolve, reject) => {
       // this function is in a repeating callback that terminates when
       // the promise is resolved
 
       // TODO: use react-query to refetch this instead of the custom repeater
       if (!this.props.app) {
-        return;
+        resolve();
```

Comment:
this was a little confusing but hopefully it passes the tests. if you define `resolve` inside of a promise you have to use it instead of return.